test/elementhandle-convenience.spec.ts

```diff
@@ -176,6 +176,22 @@ it('isVisible and isHidden should work', async ({ page }) => {
   expect(await page.isHidden('no-such-element')).toBe(true);
 });
 
+it('element state checks should work for label with zero-sized input', async ({page, server}) => {
+  await page.setContent(`
+    <label>
+      Click me
+      <input disabled style="width:0;height:0;padding:0;margin:0;border:0;">
+    </label>
+  `);
+  // Visible checks the label.
+  expect(await page.isVisible('text=Click me')).toBe(true);
+  expect(await page.isHidden('text=Click me')).toBe(false);
+
+  // Enabled checks the input.
```

Comment:
This inconsistency between isVisible and isEnabled is somewhat confusing. I'd at least expect them to check the same element (but I understand that it comes with another issues like you wouldn't be able to check if input is fillable).